src/main/java/org/opentripplanner/graph_builder/module/osm/OpenStreetMapModule.java

```diff
@@ -914,33 +904,76 @@ private void buildElevatorEdges(Graph graph) {
-                    }
+            // Add highway=elevators to graph as elevators
+            Iterator<OSMWay> elevators = osmdb.getWays().stream()
+                .filter(way -> way.hasTag("highway") && "elevator".equals(way.getTag("highway")) && !osmdb.isAreaWay(way.getId()))
+                .iterator();
+
+            for (Iterator<OSMWay> it = elevators; it.hasNext(); ) {
+                OSMWay elevatorWay = it.next();
 
-                    // The narrative won't be strictly correct, as it will show the elevator as part
-                    // of the cycling leg, but I think most cyclists will figure out that they
-                    // should really dismount.
-                    ElevatorHopEdge foreEdge = new ElevatorHopEdge(from, to, permission);
-                    ElevatorHopEdge backEdge = new ElevatorHopEdge(to, from, permission);
-                    foreEdge.wheelchairAccessible = wheelchairAccessible;
-                    backEdge.wheelchairAccessible = wheelchairAccessible;
+                List<Long> nodes = Arrays.stream(elevatorWay.getNodeRefs().toArray())
+                    .filter(nodeRef -> intersectionNodes.containsKey(nodeRef) && intersectionNodes.get(nodeRef) != null)
+                    .boxed().collect(Collectors.toList());
+
+                ArrayList<Vertex> onboardVertices = new ArrayList<Vertex>();
+                for (int i = 0; i < nodes.size(); i++) {
+                    Long node = nodes.get(i);
+                    OsmVertex sourceVertex = intersectionNodes.get(node);
+                    String sourceVertexLabel = sourceVertex.getLabel();
+
+                    ElevatorOffboardVertex offboardVertex = new ElevatorOffboardVertex(
+                        graph,
+                        sourceVertexLabel + "_offboard", sourceVertex.getX(),
+                        sourceVertex.getY(), elevatorWay.getId() + " / " + i
+                    );
+
+                    new FreeEdge(sourceVertex, offboardVertex);
+                    new FreeEdge(offboardVertex, sourceVertex);
+
+                    ElevatorOnboardVertex onboardVertex = new ElevatorOnboardVertex(graph,
+                        sourceVertexLabel + "_onboard", sourceVertex.getX(),
+                        sourceVertex.getY(), elevatorWay.getId() + " / " + i);
+
+                    new ElevatorBoardEdge(offboardVertex, onboardVertex);
+                    new ElevatorAlightEdge(onboardVertex, offboardVertex, elevatorWay.getId() + " / " + i);
+
+                    onboardVertices.add(onboardVertex);
                 }
-            } // END elevator edge loop
+                createElevatorHopEdges(onboardVertices, elevatorWay.isTagFalse("bicycle"), elevatorWay.isTagFalse("wheelchair"));
+                LOG.debug("Created elevatorHopEdges for way {}", elevatorWay.getId());
+            }
+        }
+
+        private void createElevatorHopEdges(ArrayList<Vertex> onboardVertices, boolean noBicycle, boolean noWheelchair) {
```

Comment:
I would turn the booleans around: `bicycleAllowed` instead of `noBicycle` and the same for wheelchair.